contributor-dashboard-legacy/functions/circle-ci-artifacts.js

```diff
@@ -0,0 +1,42 @@
+async function fetchCircleCIApiV2(endpoint) {
+	const url = `https://circleci.com/api/v2/${endpoint}`;
+
+	// eslint-disable-next-line no-console
+	console.log(url);
+	const response = await fetch(url);
+	const json = await response.json();
+	return json;
+}
+
+/**
+ * netlify function that wraps CircleCI API v2 which requires authentification.
+ *
+ * @param {*} event
+ * @param {*} context
+ */
+exports.handler = async function circleCIArtefact(event) {
```

Comment:
Bypass CORS after CircleCI API change.

Reply:
Do I understand correctly that to fill the table we're fetching from multiple functions and reconcile them client-side?
It feels somewhat wasteful to me, especially since some of the functions seem to do cache control and others don't. Isn't it more logical to fetch and reconcile data in a single netlify function, add cache control and serve it in a format that can easily be fed into a table?

Reply:
Agree

Reply:
🤔 Actually, we're already calculating the diff in the [dangerfile](https://github.com/mui/material-ui/blob/aa8a03d3def4ccc8c7a40e9ea783896ed81897c4/dangerFileContent.ts#L148C50-L186). Would probably make more sense to just calculate the diff in CI, store it on file and push it to S3 and make it public. Then danger can just read the file and the UI/toolpad can fetch from S3. We wouldn't have to fetch and calculate in 3 places and we wouldn't have the problem with disappearing expired build artifacts neither.

Not for this PR though.